src/CommunityPatch/Patches/Perks/Cunning/Tactics/ElusivePatch.cs

```diff
@@ -0,0 +1,103 @@
+    private static readonly MethodInfo SacrificePatchMethodInfo = typeof(ElusivePatch).GetMethod(nameof(SacrificePostfix), Public | NonPublic | Static | DeclaredOnly);
+
+    public override IEnumerable<MethodBase> GetMethodsChecked() {
+      yield return SacrificeTargetMethodInfo;
+      yield return BreakOutTargetMethodInfo;
+      yield return BreakInTargetMethodInfo;
+    }
+
+    private PerkObject _perk;
+
+    private static readonly byte[][] SacrificeHashes = {
+      new byte[] {
+        // e1.2.1.226961
+        0x36, 0x69, 0x6E, 0xC5, 0x43, 0xA7, 0xF4, 0x0F,
+        0x0A, 0x98, 0xB6, 0xC2, 0x96, 0x63, 0x43, 0x24,
+        0x5A, 0xC4, 0x4A, 0x65, 0x28, 0x72, 0xB0, 0x59,
+        0xEF, 0x1B, 0x96, 0x03, 0xE2, 0xBE, 0x32, 0x21
+      }
+    };
+    
+    private static readonly byte[][] BreakOutHashes = {
+      new byte[] {
+        // e1.2.1.226961
+        0x0A, 0x4A, 0x41, 0xC1, 0xA6, 0x01, 0xC1, 0x8B,
+        0x9A, 0xD8, 0x0D, 0xFD, 0x6B, 0x1E, 0x6A, 0x6C,
+        0x9A, 0x1B, 0x15, 0x14, 0xCF, 0x55, 0x28, 0xF0,
+        0x44, 0x2B, 0x75, 0xE3, 0x7B, 0xD7, 0xA6, 0xEA
+      }
+    };
+    
+    private static readonly byte[][] BreakInHashes = {
+      new byte[] {
+        // e1.2.1.226961
+        0x16, 0x46, 0x40, 0x13, 0x98, 0x24, 0x44, 0xB0,
+        0xDD, 0x70, 0xD0, 0x88, 0xF8, 0x2B, 0x22, 0x23,
+        0x49, 0x2F, 0x2D, 0x41, 0x0D, 0xA6, 0xF1, 0x94,
+        0xDA, 0x0B, 0xA9, 0xFD, 0xDB, 0xC2, 0x3F, 0xC4
+      }
+    };
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "iZ5hXYTl");
+
+    public override bool? IsApplicable(Game game) {
+      if (_perk == null) return false;
+
+      return IsTargetPatchable(SacrificeTargetMethodInfo, SacrificeHashes) &&
+        IsTargetPatchable(BreakOutTargetMethodInfo, BreakOutHashes) && 
+        IsTargetPatchable(BreakInTargetMethodInfo, BreakInHashes);
+    }
+
+    public override void Apply(Game game) {
+      if (Applied) return;
+      CommunityPatchSubModule.Harmony.Patch(SacrificeTargetMethodInfo, postfix: new HarmonyMethod(SacrificePatchMethodInfo));
+      CommunityPatchSubModule.Harmony.Patch(BreakOutTargetMethodInfo, postfix: new HarmonyMethod(SiegePatchMethodInfo));
+      CommunityPatchSubModule.Harmony.Patch(BreakInTargetMethodInfo, postfix: new HarmonyMethod(SiegePatchMethodInfo));
+      Applied = true;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void SiegePostfix(ref int __result, MobileParty party)
+      => ApplyPerk(ref __result, party);
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void SacrificePostfix(ref int __result, BattleSideEnum battleSide, MapEvent mapEvent) {
+      var side = battleSide == BattleSideEnum.Attacker ? mapEvent.AttackerSide : mapEvent.DefenderSide;
+      
+      foreach (var sideParty in side.Parties.Where(x => x.MobileParty != null))
+        ApplyPerk(ref __result, sideParty.MobileParty);
```

Comment:
Idk this might be a bit overpowered. It seems that if e.g. 2 party leaders in the side have this perk, the men sacrificed for the entire side is reduced by 50% and then 50% again, even if they only command a small fraction of the side's total men.

Maybe it would make more sense to have the perk's reduction be applied once and scaled by the proportion of men that are led by a leader with the perk?

```suggestion
      __result -= __result * percentMenLeadByLeaderWithPerk * ( 1.0f - perk.PrimaryBonus )
```


( just did `( 1 - bonus )` because the description of the perk is 50% fewer men with a value of 0.5f. So if they someday change it to 30% fewer men I'd say they'd set the bonus to be 0.3f so we'd have to 1 - 0.3 before mult. Admittedly this is a guess. )

Reply:
Yeah it's OP when stacking.

Reply:
In addition maybe if the army's leader has the perk then the bonus could be gotten for the entire army regardless of the individual party leaders. That's how TW does it for the Assuring Presence perk (which is 20% less army cohesion decay but has the role PartyLeader).

But otherwise fall back on it being based on the percentage of men who are in qualifying parties? This is just speculative but that's how I would do it.

Example of Assuring Presence perk from `CalculateCohesionChangeInternal`:

```CSharp
if (!army.LeaderParty.HasPerk(DefaultPerks.Leadership.AssuringPresence))
        return;
float perkMultiplyFactor = DefaultPerks.Leadership.AssuringPresence.PrimaryBonus * 0.01f;
float cohesionIncrease = Math.Abs(
    cohesionChange.ResultNumber  // negative
        * perkMultiplyFactor);
cohesionChange.Add(cohesionIncrease, DefaultArmyManagementCalculationModel._assuringPresencePerkText);
```

Reply:
Possible implementation of my suggestion:

iPherian/MnB2-Bannerlord-CommunityPatch@25d19399894823bb0b54a3a15f23c83a98ed885c

Reply:
I was taking a look in your implementation and it looks great. I'll give some test to make sure it is working fine, then I'll let you know.

Reply:
There was a bug with the implementation where it wasn't subtracting the sacrifice reduction from sacrifice total, but that was a bug already in my implementation too. Also, I refactored your code just a little bit according to ReSharper suggestions. 

I did some test and this is working fine. Thank you so much iPherian for suggesting and comming up with the solution!

Reply:
Nice catch!